tests/system_test.py

```diff
@@ -113,6 +114,39 @@ def _check_requirements():
 MISSING_REQUIREMENTS = _check_requirements()
 
 
+def flakey(issue: str, repeats: int = 3):
+    """Decorator that marks test as flakey (flaky).
+
+    If applied, executes the test up to three times, marking it as failed only if it fails each time.
+    Note, that use of this decorator is generally discouraged---tests should pass reliably when their assertions are upheld.
+
+    Example usage
+
+        @flakey(issue='#123')
+        def test_feature_in_fragile_manner():
+            self.assertTrue(...)
+    """
+    del issue  # unused
+
+    def decorator(f):
+        @functools.wraps(f)
+        def inner(*args, **kwargs):
+            # Python 3 clears the exception upon leaving the `except` block
+            # https://docs.python.org/3/reference/compound_stmts.html#the-try-statement
+            preserved_exc = None
+            for _ in range(repeats):
+                try:
+                    return f(*args, **kwargs)
+                except AssertionError as exc:
+                    preserved_exc = exc
+                    logging.info("Test marked as flaky has failed: %s", exc)
+            raise AssertionError("Flaky test has failed too many times") from preserved_exc
```

Comment:
Ken seems to use `flakey`, I am used to seeing `flaky`. I'm going to use the Ken's terminology thoughout.